Rpg sheet - Add components and custom hook

julialbq opened the pull request:
Add components to create a character sheet

- RpgForm is a form that recieves input from the user to fill the information about the character
- Identification deals with the character's name and last name
- Life deals with the character's life and mana information
- Level deals with level, experience and armor information

UseInput is a custom hook to handle the input values

julialbq committed:
chore: add character classes images

julialbq committed:
feat: add RpgForm component

julialbq committed:
feat: add Level, Life and Identification components

julialbq committed:
feat: add custom hook UseInput

julialbq committed:
refactor: add useInput logic to App and pass props to components

julialbq committed:
refactor: add handleSubmit function to form - save in localStorage

julialbq committed:
style: add style to components Identification, Level and Life and App

julialbq committed:
refactor: add logic to stablish maximum life points

julialbq committed:
refactor: add logic to establish armor points

julialbq committed:
feat: add custom hook useLocalStorage and its implementation in RpgForm component and App

julialbq committed:
delete unnecessary files

julialbq committed:
style: add style to RpgForm and Identification components and reset style for button, input and select tags

julialbq committed:
chore: change icon images to public folder

julialbq committed:
chore: small changes do Identification (still needs to finish rendering imagens) and RpgForm components)

julialbq committed:
feat: add class icon to the Identification component

julialbq committed:
style: improve form style in RpgForm component

julialbq committed:
style: edit Level and Life components style (width)

julialbq committed:
fix: fix localStorage key (had set wrong variable to it)

julialbq commented:
![rpg-sheet](https://user-images.githubusercontent.com/93345010/218116158-1e7e2feb-7d14-44cb-a2e9-2914ed045ce3.gif)

julialbq commented:
useLocalStorare is a customHook that deals with saving, reseting and retrieving information from local storage

julialbq committed:
refactor: add function resetAll in RpgForm component

julialbq committed:
style: add border to form according to the chosen character class

laka3000 commented on src/App.css:
```diff
@@ -1,38 +1,26 @@
-    animation: App-logo-spin infinite 20s linear;
-  }
+.icon {
+  position: absolute;
```
Pelo que foi apresentado, não tinha necessidade de usar position absolute aqui, um display: flex resolveria.

laka3000 commented on src/components/CustomHooks/useLocalStorage.jsx:
```diff
@@ -0,0 +1,20 @@
+
+export const useLocalStorage = (key, defaultValue) => {
+
+  const value = useState(() => {
```
Hook não está sendo usado corretamente aqui, o `useState` retorna:
`[valorAtual, funçãoQueAtualizaOValor]`, e o argumento passado para o hook é o valor inicial desejado.

laka3000 commented on src/components/Level/Level.js:
```diff
@@ -0,0 +1,20 @@
+export const Level = ({level, xp, life}) => {
+  const [armor, setArmor] = useState(0)
+  
+  useMemo(() => {
```
Hook também não usado corretamente aqui, está sendo usado como um https://reactjs.org/docs/hooks-effect.html

laka3000 commented on src/components/RpgFom/RpgForm.jsx:
```diff
@@ -0,0 +1,128 @@
+
+    const elements = event.target.elements
+
+    formValues = {
```
Essa linha não teria problema se estivesse prefixada por `const`, como `const formValues = `, aqui o `formValues` passado como prop está sendo sobrescrito.

julialbq committed:
refactor: reviewer changes

julialbq committed:
style: add option for transparent border when no class is selected (used when the reset function is called)

julialbq committed:
refactor: change Life component to add buttons for increase/decrease of life and mana, also added 'armor' because removed the Level component

julialbq commented:
![rpg-sheet-2](https://user-images.githubusercontent.com/93345010/218539371-3324ec83-7d19-433a-b5f8-61f3f7c0319b.gif)

- Refactor of useLocalStorageHook
- Added buttons for Mana and Life
- Removed the Level component (rendering only what was asked in the challenge)
- Added 'armor' to the Life component
